### PR TITLE
enhancement/date-picker

### PR DESCRIPTION
**Pull request by nekosaur**
- Removed props `days`, `shortDays` , `months`
- Removed unused code
- Changed props `title-date-format` and `header-date-format` to accept an object in the style of the options argument of [Date.toLocaleString]
(https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString)
- Changed `first-day-of-week` to accept String and Number. The value represents the day the week starts and begins with Sunday as 0.
- Added `locale` prop that take a BCP 47 language tag ('en-us', 'zh-cn', etc)
- Fixed bug reported in #894 
- Related to #822

Updated example found in https://github.com/vuetifyjs/docs/pull/107

### CODE REVIEW

File: src/components/pickers/DatePicker.js [hunk 1]
```diff
@@ -19,53 +20,32 @@ export default {
...
+      isReversing: false,
+      narrowDays: [],
+      firstDayOffset: 0
     }
   },
 
   props: {
+    locale: {
+      type: String,
+      default: 'en-us'
```
Comment (mfferreira) [inline src/components/pickers/DatePicker.js]: Can you make the default the users current locale?

Comment (nekosaur) [inline src/components/pickers/DatePicker.js]: When I've previously had to detect local browser locale the solutions have been pretty hacky. But I can take another look.

Comment (mfferreira) [inline src/components/pickers/DatePicker.js]: Using something like `navigator.languages` > `navigator.language`

**Comment by johnleider** [comment]
Please refer to the issues I posted here: https://github.com/vuetifyjs/docs/pull/107

File: src/components/pickers/DatePicker.js (hunk 1 above)
Comment (richardtallent) [inline src/components/pickers/DatePicker.js]: Here's the code I've been using in vue-simple-calendar, which uses the locale and Intl for the day and month names:

```
return (
	(navigator.languages && navigator.languages.length) ?
		navigator.languages[0]
		: navigator.language || navigator.browserLanguage
	).toLowerCase();
```

File: src/components/pickers/DatePicker.js [hunk 2]
```diff
@@ -19,60 +20,38 @@ export default {
...
-      ]
-    },
     allowedDates: {
       type: [Array, Object, Function],
       default: () => (null)
     },
     firstDayOfWeek: {
-      type: String,
-      default: 'Sunday'
+      type: Number,
```
Comment (johnleider) [inline src/components/pickers/DatePicker.js]: Should make this [Number, String] so the user doesn't have to bind a prop in order to change this

File: src/components/pickers/mixins/date-table.js [hunk 3]
```diff
@@ -37,8 +37,8 @@ export default {
     genTHead () {
       return this.$createElement('thead', {
 
-      }, this.genTR(this.week.map((o, i) => {
-        return this.$createElement('th', Array.isArray(this.shortDays) && this.shortDays[i] || this.shortDays(o))
+      }, this.genTR(this.narrowDays.map((day) => {
```
Comment (johnleider) [inline src/components/pickers/mixins/date-table.js]: remove parenthesis, (day) => to day =>

File: src/components/pickers/mixins/date-table.js [hunk 4]
```diff
@@ -53,7 +53,7 @@ export default {
       const day = new Date(
         this.tableYear,
         this.tableMonth
-      ).getDay() - this.days.indexOf(this.firstDayOfWeek)
+      ).getDay() - this.firstDayOfWeek
```
Comment (johnleider) [inline src/components/pickers/mixins/date-table.js]: since we are accepting firstDayOfWeek as a string, just do parseInt(this.firstDayOfWeek)

**Pull request merged**